File: 02-medium/1089-rotate-array/solution.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        original = list(nums)
        for index, element in enumerate(original):
            new_index = (index + k) % len(nums)
            nums[new_index] = element

File: 02-medium/1089-rotate-array/solution.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        vector<int> original = nums;

        for (int i = 0, l = nums.size(); i < l; i++) {
            int new_index = (i + k) % l;
            nums[new_index] = original[i];
        }
    }
};
